decide/src/main/java/org/example/Decide.java

```diff
@@ -37,8 +37,26 @@ public Boolean[] getConditionMetVector() {
         };
     }
 
+    public Boolean[][] computePreliminaryUnlockingMatrix(Boolean[] conditionMetVector) {
+        Boolean[][] preliminaryUnlockingMatrix = new Boolean[15][15];
+        for (int i = 0; i < 15; i++) {
+            for (int j = 0; j <= i; j++) {
+                boolean b = false;
+                switch (settings.LCM[i][j]) {
```

Comment:
(Not important) A cool thing you can do here is actually assign b to the switch, that would look something like this:
```java
boolean b = switch (settings.LCM[i][j]) {
    case NOT_USED -> true;
    case AND -> conditionMetVector[i] && conditionMetVector[j];
    case OR -> conditionMetVector[i] || conditionMetVector[j];
}
```

Reply:
Yes good thinking, doing that thanks!